src/main/java/com/saucelabs/bamboo/sod/action/BuildConfigurator.java

```diff
@@ -120,8 +120,7 @@ public void println(String x) {
         SauceTunnelManager sauceTunnelManager = SauceConnectFourManagerSingleton.getSauceConnectFourTunnelManager();
         String options = getResolvedOptions(config.getSauceConnectOptions());
         if (config.useGeneratedTunnelIdentifier()) {
-            String tunnelIdentifier = generateTunnelIdentifier(buildContext.getPlanName());
-            customVariableContext.addCustomData(SODKeys.TEMP_TUNNEL_ID, tunnelIdentifier);
+            String tunnelIdentifier = customVariableContext.getVariables(buildContext).get(SODKeys.TUNNEL_ID);
             options = "--tunnel-identifier " + tunnelIdentifier + " " + options;
```

Comment:
Actually we are passing this to ci-sauce which is internally maintaining a map too, so i think it is fine just to pass unique tunnel identifier to ci-sauce options